### PR TITLE
[OCI] 1. Support specify OS with custom image id. 2. Corner case fix

### PR DESCRIPTION
**Pull request by HysunHe**
1. Support specify OS type when specify custom image with --image-id. (Currently ubuntu/oraclelinux)  
2. A corner case fix when ctrl+c during creating cluster, which might cause the cluster cannot be terminated if the vNIC is not attached to the provisioning instance.

I'll create a doc PR later to add image-id description for OCI in task yaml.

sky launch -c img-ol8 --cloud oci --region ap-seoul-1 --cpus 2 --image-id ocid1.image.oc1.ap-seoul-1.aaaaaaaarhw7elbcsnsxmtetmicg5eh5yf5yodmjozhxzrg7h4glptqfiwua:oraclelinux "cat /etc/os-release"

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
- [x] All smoke tests: `pytest tests/test_smoke.py` 
- [x] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [x] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

**Comment by HysunHe** [comment]
@cblmemo A corner case fix and a small yet useful enhancement. could you please help do this review?

File: sky/clouds/oci.py [hunk 1]
```diff
@@ -297,15 +305,18 @@ def make_deploy_resources_variables(
...
-        image_str = self._get_image_str(image_id=resources.image_id,
-                                        instance_type=resources.instance_type,
-                                        region=region.name)
-
-        # pylint: disable=import-outside-toplevel
-        from sky.clouds.service_catalog import oci_catalog
-        os_type = oci_catalog.get_image_os_from_tag(tag=image_str,
-                                                    region=region.name)
-        logger.debug(f'OS type for the image {image_str} is {os_type}')
+        if not os_type:
```
Comment (cblmemo) [inline sky/clouds/oci.py:308]:
```suggestion
        if os_type is None:
```

File: sky/provision/oci/query_utils.py [hunk 2]
```diff
@@ -506,8 +506,11 @@ def find_nsg(cls, region: str, nsg_name: str,
             raise exceptions.ResourcesUnavailableError(
                 'The VCN is not available')
 
-        # Get the primary vnic.
-        assert len(list_vcns_resp.data) > 0
+        # Get the primary vnic. The vnic might be none for the
+        # corner case when the cluster was exited during provision.
+        if not list_vcns_resp.data:
```
Comment (cblmemo) [inline sky/provision/oci/query_utils.py:511]:
```suggestion
        if list_vcns_resp.data is None:
```

**Review by cblmemo — APPROVED** [review]
Thanks for the fix @HysunHe ! LGTM except for the nits ;)

File: sky/provision/oci/query_utils.py (hunk 2 above)
Comment (HysunHe) [reply sky/provision/oci/query_utils.py:511]: The list_vcns_resp.data would be [] here.

File: sky/provision/oci/query_utils.py [hunk 3]
```diff
@@ -506,8 +506,11 @@ def find_nsg(cls, region: str, nsg_name: str,
             raise exceptions.ResourcesUnavailableError(
                 'The VCN is not available')
 
-        # Get the primary vnic.
-        assert len(list_vcns_resp.data) > 0
+        # Get the primary vnic. The vnic might be none for the
```
Comment (cblmemo) [inline sky/provision/oci/query_utils.py:509]:
```suggestion
        # Get the primary vnic. The vnic might be an empty list for the
```
